lib/v2/screens/profile_screens/security/security_screen.dart

```diff
@@ -22,9 +24,37 @@ class SecurityScreen extends StatelessWidget {
     return Scaffold(
       appBar: AppBar(title: Text('Security'.i18n)),
       body: BlocProvider(
-        create: (context) => SecurityBloc(),
+        create: (context) =>
+            SecurityBloc(settingsNotifier: SettingsNotifier.of(context), authNotifier: AuthNotifier.of(context))
+              ..add(const SetUpInitialValues()),
         child: MultiBlocListener(
           listeners: [
+            BlocListener<SecurityBloc, SecurityState>(
+              listenWhen: (previous, current) => previous.isSecurePin == false && current.isSecurePin == true,
```

Comment:
this could be a page command. 

I think ill make it required to use pageCommands. inside `BlocListener` Its makes it much cleaner to see what we are trying to do.

Reply:
It is not clear to me what you mean by this pageCommad is an enum? and how it would works?

Reply:
look at my open pr

Reply:
I just think here I could not use a `class` and reset to `null` because the `copyWith` does not accept `null` values

Reply:
but we could make it so it takes null. 
on my next pr i need this 

class ShowTransactionSuccess {
  final String amount;
  final String toImage;
  final String toName;
  final String toAccount;
  final String fromImage;
  final String fromName;
  final String fromAccount;
  final String transactionId;

  ShowTransactionSuccess({
    @required this.amount,
    this.toImage,
    this.toName,
    @required this.toAccount,
    this.fromImage,
    this.fromName,
    @required this.fromAccount,
    @required this.transactionId,
  });

  ShowTransactionSuccess.withoutServerUserData({
    String amount,
    String toAccount,
    String fromAccount,
    String transactionId,
  }) : this(amount: amount, toAccount: toAccount, fromAccount: fromAccount, transactionId: transactionId);
}

If the command is an enum, then we cant pass all this data we need in the UI

Reply:
Usually in these cases I use 2 variables in the bloc state, one for the listener and the other for the data.
```
Class SomeState {
 final bool isChanged;
 final SomeModel model;
 ... 
```
That can be retrived inside the listener  `listener: (context, state)` , but I see that with your approach a variable of the state of the bloc is saved. Am I right?